index.bs

```diff
@@ -1715,3 +1705,220 @@ When [=a browsing context is discarded=], run the following steps:
  1. [=Emit an event=] with |body| and |parent id|.
 
 </div>
+
+## Script ## {#module-script}
+
+### Definition ### {#module-script-definition}
+
+[=remote end definition=] and [=local end definition=]
+```
+Realm = text;
+
+RealmOrBrowsingContext = (
+  {realm: Realm} //
+  {context: BrowsingContext}
+)
+
+RealmInfo = {
+  realm: Realm,
+  type: RealmType,
+  origin: text,
+}
+
+RealmType = (
+  "window" / "worker" / "sharedworker" / "serviceworker" / "worklet" / text
```

Comment:
Since specific types of workers are enumerated, should different worklets also not be their own types?

I'm assume that "worker" here means only dedicated workers, it's not a generic type. Should we promise in a note that all workers will always end with the string "worker" and nothing else ever will, so that clients can test if something is any kind of worker without getting nervous?

The same setup should then be OK for worklets too.

Reply:
I don't really know how to get a list of all defined worklet types, but I agree having some convention around the naming of subtypes will help.